clap_generate/tests/completions/fish.rs

```diff
@@ -47,6 +82,36 @@ fn fish_with_special_help() {
     common::<Fish>(&mut app, "my_app", FISH_SPECIAL_HELP);
 }
 
+fn build_app_special_help() -> App<'static> {
```

Comment:
Why are these functions duplicated?

* Please rename `tests/completions/mod.rs` to `tests/common.rs`
* Move these functions back to that file
* Move all these `fish.rs`, `bash.rs` etc to `tests` folder
* Delete `completions` folder
* Add `mod common` to all these `fish.rs`, `bash.rs` etc

Reply:
Regarding the code duplication. The commit message of my first commit explains the change:

```
Right now, adjusting test cases to cover some special edge-cases for a
specific shell is a little problematic.

For instance, if you add a special character that needs escaping
to one of the help strings, not only the tests for the shell you're
currently working on need adjustments, but rather all shell tests!

Contributors can often only work on one specific shell. The
need to adjust shells they don't know anything about discourages
testing or makes it simply impossible for a single contributor.

Arguably, this commit introduces a lot of code duplication, but on
the other hand it also makes testing a lot easier.
```